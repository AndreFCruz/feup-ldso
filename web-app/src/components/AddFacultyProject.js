import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';

class AddFacultyProject extends Component {

    constructor(props) {
        super(props);

        this.state = {
            title: '',
            content: ''
        };

        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleInputChange(event) {
        this.setState({
            [event.target.name]: event.target.value
        });
    }

    handleSubmit(event){
        event.preventDefault();
        const route = process.env.REACT_APP_ENDPOINT + 'faculties/en/' + this.props.match.params.faculty.toLowerCase() + '/social-projects';
        fetch(route, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(this.state),
        })
            .then(response => response.json());

        this.props.onChildSetRefresh();

    }

    render() {
        return(
            <Form>
                <FormGroup>
                    <Label for="projectTitle">Project Title</Label>
                    <Input
                        name="title"
                        placeholder="Title"
                        value={this.state.title}
                        onChange={this.handleInputChange}/>
                </FormGroup>
                <FormGroup>
                    <Label for="projectDescription">Project Description</Label>
                    <Input
                        type="textarea"
                        name="content"
                        placeholder="Project Description"
                        value={this.state.content}
                        onChange={this.handleInputChange}/>
                </FormGroup>
                <Button color="secondary" onClick={this.handleSubmit}>Add</Button>
            </Form>
        );
    }

}

AddFacultyProject.propTypes = {
    match: PropTypes.object.isRequired,
    onChildSetRefresh: PropTypes.func.isRequired
};

export default AddFacultyProject;
