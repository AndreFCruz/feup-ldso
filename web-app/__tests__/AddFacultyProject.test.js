import React from 'react';
import renderer from 'react-test-renderer';
import AddFacultyProject from '../src/components/AddFacultyProject';
import 'isomorphic-fetch'; //components use fetch() on mount
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { shallow } from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

var mockAxios = new MockAdapter(axios);

var props = {
    match: {
        params: {
            faculty: 'faup',
            project: '1'
        }
    }
};

describe('AddFacultyProject Snapshots', () => {
    it('AddFacultyProject.handleSubmit()', () => {

        mockAxios.onPost('http://localhost:3000/faculties/en/faup/social-projects').reply(200, {
            title: '[EN] Test title',
            content: '[EN] Place holder text, just a test.',
        });

        const mockupCall = jest.fn();
        const component = renderer.create(
            <AddFacultyProject
                {...props}
                onChildSetRefresh={mockupCall}
            />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Test click handle submit event', () => {

        const mockCallBack = jest.fn();

        const button = shallow((
            <AddFacultyProject
                {...props}
                onChildSetRefresh={mockCallBack}
            />));
        button.find('Button').simulate('click',
            {
                preventDefault: () => { }
            });
        expect(mockCallBack.mock.calls.length).toEqual(1);

    });
});
