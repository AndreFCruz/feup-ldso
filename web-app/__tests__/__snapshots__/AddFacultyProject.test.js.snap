// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFacultyProject Snapshots AddFacultyProject.handleSubmit() 1`] = `
<form
  className=""
>
  <div
    className="position-relative form-group"
  >
    <label
      className=""
      htmlFor="projectTitle"
    >
      Project Title
    </label>
    <input
      className="form-control"
      name="title"
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value=""
    />
  </div>
  <div
    className="position-relative form-group"
  >
    <label
      className=""
      htmlFor="projectDescription"
    >
      Project Description
    </label>
    <textarea
      className="form-control"
      name="content"
      onChange={[Function]}
      placeholder="Project Description"
      value=""
    />
  </div>
  <button
    aria-label={null}
    className="btn btn-secondary"
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
</form>
`;
